src/csharp/Microsoft.Spark/Sql/GenericRow.cs

```diff
@@ -0,0 +1,168 @@
+        ///// Returns the column value whose column name is given.
+        ///// </summary>
+        ///// <param name="columnName">Column name to look up</param>
+        ///// <returns>A column value</returns>
+        //public object Get(string columnName) =>
+        //    Get(Schema.Fields.FindIndex(f => f.Name == columnName));
+
+        /// <summary>
+        /// Returns the string version of this row.
+        /// </summary>
+        /// <returns>String version of this row</returns>
+        public override string ToString()
+        {
+            var cols = new List<string>();
+            foreach (object item in Values)
+            {
+                cols.Add(item?.ToString() ?? string.Empty);
+            }
+
+            return $"[{(string.Join(",", cols.ToArray()))}]";
+        }
+
+        /// <summary>
+        /// Returns the column value at the given index, as a type T.
+        /// TODO: If the original type is "long" and its value can be
+        /// fit into the "int", Pickler will serialize the value as int.
+        /// Since the value is boxed, <see cref="GetAs{T}(int)"/> will throw an exception.
+        /// </summary>
+        /// <typeparam name="T">Type to convert to</typeparam>
+        /// <param name="index">Index to look up</param>
+        /// <returns>A column value as a type T</returns>
+        public T GetAs<T>(int index) => (T)Get(index);
+
+        ///// <summary>
+        ///// Returns the column value whose column name is given, as a type T.
+        ///// TODO: If the original type is "long" and its value can be
+        ///// fit into the "int", Pickler will serialize the value as int.
+        ///// Since the value is boxed, <see cref="GetAs{T}(string)"/> will throw an exception.
+        ///// </summary>
+        ///// <typeparam name="T">Type to convert to</typeparam>
+        ///// <param name="columnName">Column name to look up</param>
+        ///// <returns>A column value as a type T</returns>
+        //public T GetAs<T>(string columnName) => (T)Get(columnName);
+
+        /// <summary>
+        /// Checks if the given object is same as the current object.
+        /// </summary>
+        /// <param name="obj">Other object to compare against</param>
+        /// <returns>True if the other object is equal.</returns>
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj is Row otherRow)
+            {
+                return Values.SequenceEqual(otherRow.Values);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Returns the hash code of the current object.
+        /// </summary>
+        /// <returns>The hash code of the current object</returns>
+        public override int GetHashCode() => base.GetHashCode();
+
+        //TODO:
```

Comment:
What's the plan for `Convert()`?

Reply:
I believe this requires explicit type checking for every value type and corresponding casting functions in each DataType, need confirmation @imback82 , @suhsteve ?